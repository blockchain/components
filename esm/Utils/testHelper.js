import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}