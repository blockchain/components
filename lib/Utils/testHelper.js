"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shallowWithTheme = shallowWithTheme;
exports.mountWithTheme = mountWithTheme;

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _styledComponents = require("styled-components");

var _theme = _interopRequireDefault(require("../Tools/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(_react.default.createElement(_styledComponents.ThemeProvider, {
    theme: _theme.default
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.shallow].concat(Array.prototype.slice.call(arguments)));
}

function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.mount].concat(Array.prototype.slice.call(arguments)));
}