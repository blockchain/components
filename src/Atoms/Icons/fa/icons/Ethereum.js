
  import React from 'react'
  import withStyle from '../../withStyle'

  const Ethereum = () => (
    <React.Fragment>
      <path d="M311.9 260.8L160 353.6 8 260.8 160 0l151.9 260.8zM160 383.4L8 290.6 160 512l152-221.4-152 92.8z"/>
    </React.Fragment>
  )

  export default withStyle(Ethereum,320,512)
