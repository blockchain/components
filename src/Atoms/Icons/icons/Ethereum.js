import React from 'react'

const Ethereum = (props) => (
  <svg height="1em" viewBox="0 0 32 32" width="1em" {...props}>
    <path d="M6 18.144a1.586 1.586 0 0 1-.189-1.355l-.003.011C9.168 11.2 12.592 5.64 16 .08c3.408 5.56 6.832 11.12 10.192 16.72a1.596 1.596 0 0 1-.188 1.374l.004-.006C22.704 22.784 19.344 27.384 16 32c-3.344-4.616-6.696-9.216-10-13.856zm9.6 3.312V3.296C12.544 8.28 10.48 12 7.552 16.8c3 1.768 5.04 2.928 8.024 4.656zm8.8-4.52c-2.976-4.888-4.616-8.648-7.664-13.648V21.6c2.968-1.752 4.592-2.912 7.624-4.664zm-9.504 6.096c-2.544-1.512-4.496-2.712-7.096-4.2 2.64 3.64 4.944 6.456 7.608 10.144 0-1.84.072-3.2 0-5.072a1.247 1.247 0 0 0-.508-.869l-.004-.003zm1.904 5.96c2.544-3.528 4.8-6.76 7.376-10.352-.312.208-4.656 3.112-7.04 4.552a1.18 1.18 0 0 0-.32.884v-.003c-.064 1.968-.016 2.952-.016 4.92z" />
  </svg>
)

export default Ethereum
