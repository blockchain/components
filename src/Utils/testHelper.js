import React from 'react'
import { shallow, mount } from 'enzyme'
import { ThemeProvider } from 'styled-components'
import theme from '../Tools/theme'

function wrapWithTheme(fn, children, options) {
  const wrapper = fn(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>,
    options,
  )

  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext(),
  })
}

export function shallowWithTheme() {
  return wrapWithTheme(shallow, ...arguments)
}

export function mountWithTheme() {
  return wrapWithTheme(mount, ...arguments)
}
