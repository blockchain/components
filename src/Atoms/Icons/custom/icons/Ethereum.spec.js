
  import React from 'react'
  import { mountWithTheme } from '../../../../Utils/testHelper'

  import Ethereum from './Ethereum'

  describe('Ethereum', () => {
    it('should render correctly', () => {
      const tree = mountWithTheme(<Ethereum />)
      expect(tree.find(Ethereum)).toMatchSnapshot()
    })
  })
