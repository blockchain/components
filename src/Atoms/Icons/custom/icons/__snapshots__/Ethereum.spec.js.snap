// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ethereum should render correctly 1`] = `
.c0 {
  fill: #545456;
  cursor: default;
}

.c0:hover {
  fill: #545456;
}

<withStyle
  selectable={false}
  size="2.75rem"
>
  <styled.svg
    height="2.75rem"
    selectable={false}
    viewBox="0 0 24 24"
    width="2.75rem"
    xmlns="http://www.w3.org/2000/svg"
  >
    <svg
      className="c0"
      height="2.75rem"
      viewBox="0 0 24 24"
      width="2.75rem"
      xmlns="http://www.w3.org/2000/svg"
    >
      <Ethereum>
        <path
          d="M15.9634418,11.5939634 L12.0477432,13.6478025 C12.0186645,13.6630547 11.9839508,13.6630594 11.9548679,13.647815 L8.03657207,11.5939657 C7.98765618,11.5683255 7.9687875,11.5078859 7.99442769,11.45897 C7.99540947,11.457097 7.99645042,11.4552556 7.99754896,11.4534486 L11.915832,5.00819127 C11.9445216,4.9609992 12.0060358,4.94599999 12.0532279,4.9746896 C12.0669259,4.9830171 12.0784212,4.99451574 12.0867447,5.00821623 L16.0024561,11.4534837 C16.0311319,11.5006842 16.0161147,11.562194 15.9689143,11.5908698 C15.967123,11.5919581 15.9652979,11.5929898 15.9634418,11.5939634 Z M8.14579229,12.714348 L11.9169872,18.3232742 C11.9478027,18.3691064 12.0099379,18.3812798 12.0557701,18.3504643 C12.066507,18.3432452 12.0757411,18.3340112 12.0829601,18.3232742 L15.854155,12.714348 C15.8849705,12.6685159 15.8727971,12.6063806 15.826965,12.5755651 C15.7976538,12.5558576 15.7601205,12.5530742 15.7282231,12.5682427 L11.9999736,14.3411716 L8.27172418,12.5682427 C8.22184797,12.5445246 8.162188,12.56573 8.1384699,12.6156062 C8.12330143,12.6475036 8.12608478,12.6850369 8.14579229,12.714348 Z"
          id="Ethereum"
        />
      </Ethereum>
    </svg>
  </styled.svg>
</withStyle>
`;
